### DISCUSSION TITLE
atomsWithQuery goes wrong if unmount while fetching

### DISCUSSION THREAD

**Discussion by mattp0123** [pmndrs/jotai#2039] [Q&A]
https://codesandbox.io/p/sandbox/tanstack-query-problem-forked-4fzr42?file=%2Fsrc%2FApp.tsx%3A20%2C30
In this sandbox, the `todosLoadable.state` will be `loading` permanently.

**Comment by Aslemammad** [comment]
I guess https://github.com/jotaijs/jotai-tanstack-query/pull/38 would solve the issue.
